documents/anchor.go

```diff
@@ -18,6 +20,7 @@ type AnchorProcessor interface {
 	PreAnchorDocument(ctx context.Context, model Model) error
 	AnchorDocument(ctx context.Context, model Model) error
 	SendDocument(ctx context.Context, model Model) error
+	RequestDocumentWithToken(ctx context.Context, requesterID identity.DID, tokenIdentifier, entityIdentifier, entityRelationIdentifier []byte) (*p2ppb.GetDocumentResponse, error)
```

Comment:
I agree the processor should handle the request

Reply:
I think I prefer three methods otherwise you always need to pass a lot of empty parameters. (nft registry, etc).

We could make a more generic method in the service itself.